ToolBox/BovespaDownloader/BovespaDataDownloader.cs

```diff
@@ -0,0 +1,449 @@
+
+            #region For equity daily files, use equity daily data!
+            if (type == SecurityType.Equity && resolution == Resolution.Daily)
+            {
+                for (var year = startUtc.Year; year <= endUtc.Year; year++)
+                {
+                    foreach (var bar in GetDailyData(symbol, year))
+                    {
+                        startUtc = bar.Time;
+                        yield return bar;
+                    }
+                }
+            }
+            #endregion
+
+            foreach (var file in ListTickDataFiles(type, resolution, startUtc, endUtc))
+            {
+                // Request all ticks for a specific date
+                var ticks = GetTickData(symbol, file);
+
+                if (resolution == Resolution.Tick)
+                {
+                    foreach (var tick in ticks)
+                    {
+                        yield return tick;
+                    }
+                }
+                else
+                {
+                    foreach (var bar in AggregateTicks(symbol, ticks, resolution))
+                    {
+                        yield return bar;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Get daily bars for the specified symbol from given file.
+        /// If we do not have the file locally we download it from Dropbox
+        /// </summary>
+        /// <param name="symbol">The requested symbol</param>
+        /// <param name="date">The requested date</param>
+        /// <returns>An enumerable of ticks</returns>
+        private IEnumerable<BaseData> GetDailyData(Symbol symbol, int year)
+        {
+            var localfile = String.Format("{0}/COTAHIST_A{1}.zip", _inputDirectory, year);
+            var remotefile = String.Format("https://dl.dropboxusercontent.com/u/44311500/Data/Equity/COTAHIST_A{0}.zip", year);
+        
+            #region Download file from Dropbox if it does not exist locally
+            if (!File.Exists(localfile) || new FileInfo(localfile).Length == 0 || year == DateTime.Now.Year)
+            {
+                using (var client = new WebClient())
+                {                    
+                    try { client.DownloadFile(remotefile, localfile); }
+                    catch (Exception) { }
+                }
+            }
+            #endregion
+
+            if (!File.Exists(localfile))
+            {
+                Console.WriteLine("Do not have nor could download COTAHIST_A{0}.zip", year);
+            }
+            else
+            {
+                ZipFile zip;
+                using (var reader = Compression.Unzip(localfile, out zip))
+                {
+                    var bars = reader.ReadToEnd().Split(Environment.NewLine.ToCharArray(), StringSplitOptions.RemoveEmptyEntries)
```

Comment:
I'm always weary of calling `string.Split` on potentially large strings. [Here's](http://www.codeproject.com/Articles/176031/Out-of-Memory-Exception-A-simple-string-Split-can) a great, in-depth look into why.  If the string isn't that large then no need to change, more just for information :smile: